dandi/cli/tests/test_register.py

```diff
@@ -0,0 +1,50 @@
+import re
+from os.path import dirname
+from pathlib import Path
+from traceback import format_exception
+
+from click.testing import CliRunner
+import pytest
+import yaml
+
+from ..command import register
+from ...consts import dandiset_identifier_regex, dandiset_metadata_file
+
+
+def yaml_load(s):
+    obj = yaml.load(s, Loader=yaml.BaseLoader)
+    assert len(obj) == 1  # will be a list with a single elem
+    return obj[0]
+
+
+def show_result(r):
+    if r.exception is not None:
+        return "".join(format_exception(*r.exc_info))
+    else:
+        return r.output
+
+
+def test_smoke(local_docker):
+    runner = CliRunner(mix_stderr=False)
+    with runner.isolated_filesystem():
+        Path(dandiset_metadata_file).write_text("{}\n")
```

Comment:
If there's no `dandiset.yaml` in the current directory, `register` will print the new metadata to stdout instead of saving it in a file.  Is that the intended behavior?  If so, does it matter which route the test takes?  I'll think I'll test both routes for now.